### ISSUE TITLE
[BUG] It seems that the prediction of objective "multi:softprob" is incorrect and the order of the probability is inconsistent with mllib.

### ISSUE THREAD

**Issue by sonetto19999** [dmlc/xgboost#10646]
multi:softmax output:
<img width="662" alt="WeChatWorkScreenshot_2c4ac7c7-35ba-494f-8942-b0ba5f64027d" src="https://github.com/user-attachments/assets/3c568b27-d8f4-49f6-8492-1da75f3f278b">

multi:softprob output:
<img width="809" alt="WeChatWorkScreenshot_28a6740f-d7f8-467f-9ed1-c05d35f02b2e" src="https://github.com/user-attachments/assets/93d16db4-ef48-4bb7-b414-8540d107f417">

these two figure clearly shows that the prediction of "multi:softprob" is incorrect, the first three rows shold be 0 rathe than 1 (just like the prediction of "multi:softmax")

**Comment by trivialfis** [comment]
cc @wbo4958

**Comment by sonetto19999** [comment]
anyone help?

**Comment by mayer79** [comment]
@sonetto19999 You might expect more help with a minimally reproducible example

**Comment by wbo4958** [comment]
Sorry for delay, I just see this issue. That's terrible. seems the prediction values of softprob is reversed of softmax.

Hi @sonetto19999, were you using xgboost4j-spak or xgboost pyspark?

**Comment by sonetto19999** [comment]
> @wbo4958 currently using xgboost4j-spark_2.12

**Comment by wbo4958** [comment]
I tried the iris dataset, and I didn't see this issue. BTW, is your dataset binary or multi class?
